bftengine/src/bftengine/messages/MsgCode.hpp

```diff
@@ -54,6 +54,7 @@ class MsgCode {
     ClientBatchRequest = 750,
     ClientReply = 800,
     Reserved = 900,
+    MaxMsgCodeVal  // always keep this last in the enum as it is used to allocate array size(s)
```

Comment:
1: nit: LastMsgCodeVal makes you idea more clear (keep it last, right?)
2. you shoould static assert instead someone doesnt really care about your array which is in a completely different place. This is not theoretical - tomoreow we refactor code and decide to seperate messages in the ranges of 0x100000 then 0x200000 up to 0xfffffffffff
3. what we get is a huge bitset/array (hopefully it will crash the process).

Conclusion: after your  static members definition,  static assert on them (complication time not runtime!) that they cant be more than 1000B. if yes, write a comment above that the whole idea whould be refacotred into an std::map. (and yes, its possible but harder).